src/sentry/api/endpoints/rule_snooze.py

```diff
@@ -0,0 +1,96 @@
+class RuleSnoozeEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectAlertRulePermission,)
+
+    def post(self, request: Request, project, rule_id) -> Response:
+        if not features.has("organizations:mute-alerts", project.organization, actor=None):
+            return Response(
+                {"detail": "This feature is not available for this organization."}, status=401
+            )
+
+        serializer = RuleSnoozeSerializer(data=request.data)
+        if serializer.is_valid():
+            data = serializer.validated_data
+
+            issue_alert_rule = None
+            metric_alert_rule = None
+
+            if data.get("rule"):
+                try:
+                    issue_alert_rule = Rule.objects.get(id=rule_id)
+                except Rule.DoesNotExist:
+                    raise serializers.ValidationError("Rule does not exist")
+
+            if data.get("alert_rule"):
+                try:
+                    metric_alert_rule = AlertRule.objects.get(id=rule_id)
+                except AlertRule.DoesNotExist:
+                    raise serializers.ValidationError("Rule does not exist")
+
+            # don't allow editing of a rulesnooze object for a given rule and user (or no user)
+            if RuleSnooze.objects.filter(
+                user_id=data.get("user_id"), rule=issue_alert_rule, alert_rule=metric_alert_rule
+            ).exists():
+                raise ConflictError("RuleSnooze already exists for this rule and scope.")
+
+            RuleSnooze.objects.create(
+                user_id=data.get("user_id"),
+                owner_id=request.user.id,
+                rule=issue_alert_rule,
+                alert_rule=metric_alert_rule,
+                until=data.get("until"),
+                date_added=datetime.datetime.now(),
+            )
+            if not data.get("user_id"):
+                # create an audit log entry if the rule is snoozed for everyone
+                rule = issue_alert_rule or metric_alert_rule
+                audit_log_event = "RULE_SNOOZE" if issue_alert_rule else "ALERT_RULE_SNOOZE"
+
+                self.create_audit_entry(
+                    request=request,
+                    organization=project.organization,
+                    target_object=rule.id,
+                    event=audit_log.get_event_id(audit_log_event),
+                    data=rule.get_audit_log_data(),
+                )
+            rule_id = None
+            alert_rule_id = None
+            if issue_alert_rule:
+                rule_id = issue_alert_rule.id
+            if metric_alert_rule:
+                alert_rule_id = metric_alert_rule.id
+
+            rule_snooze = {
+                "ownerId": request.user.id,
+                "userId": data.get("user_id", "everyone"),
+                "ruleId": rule_id,
+                "alertRuleId": alert_rule_id,
+                "until": data.get("until", "forever"),
+                "dateAdded": datetime.datetime.now(),
+            }
```

Comment:
can we not simply serialize the object created above? e.g. how it's done here:

https://github.com/getsentry/sentry/blob/b67df8ce3aa7590658f3eb49afceb762dffb6bf5/src/sentry/api/endpoints/organization_teams.py#L158

```
rule_ snooze = RuleSnooze.objects.create(...)

...
serialize(rule_snooze,request.user))
```

Reply:
additionally, it seems like calling `serialize` without a serializer passed in simply returns the object. if the intention is to just return the object, I don't think you need to call `serialize` on it.

https://github.com/getsentry/sentry/blob/b67df8ce3aa7590658f3eb49afceb762dffb6bf5/src/sentry/api/serializers/base.py#L64

Reply:
Yeah now that I think about it more idk if I even need to return the data at all 🤔 I'll think about it more Monday.